lindi/LindiH5pyFile/LindiH5pyFile.py

```diff
@@ -218,7 +307,7 @@ def driver(self):
 
     @property
     def mode(self):
-        return self._mode
+        return 'r' if self._mode == 'r' else 'r+'
```

Comment:
Why is this needed?

Reply:
You would think it should return self._mode. But actually, h5py only returns r or r+. See

https://github.com/h5py/h5py/blob/b2a83a03d9850076a427685a488ceeb0f77c5bd0/h5py/_hl/files.py#L318-L323

So we need to do this to be consistent.

Reply:
That's odd. OK